restli-common/src/main/java/com/linkedin/restli/internal/common/URIMaskUtil.java

```diff
@@ -121,161 +121,161 @@ public static MaskTree decodeMaskUriFormat(StringBuilder toparse) throws Illegal
+                  }
 
-              ii++;
+                  subPosition++;
 
-              // Aggressively consume the numerical value for the range parameter as this is a special case.
-              StringBuilder rangeValue = new StringBuilder();
-              while (ii < toparse.length())
-              {
-                char ch = toparse.charAt(ii);
-                if (ch == ',')
-                {
-                  state = ParseState.TRAVERSE;
-                  nextToken = ii;
-                  break;
-                }
-                else if (ch == ')')
-                {
-                  state = ParseState.ASCEND;
-                  nextToken = ii;
-                  break;
-                }
-                else if (Character.isDigit(ch))
-                {
-                  rangeValue.append(ch);
+                  // Aggressively consume the numerical value for the range parameter as this is a special case.
+                  maskValue = 0;
+                  while (subPosition < toparse.length())
+                  {
+                    char ch = toparse.charAt(subPosition);
+                    if (ch == ',')
+                    {
+                      state = ParseState.TRAVERSE;
+                      nextToken = subPosition;
+                      break;
+                    }
+                    else if (ch == ')')
+                    {
+                      state = ParseState.ASCEND;
+                      nextToken = subPosition;
+                      break;
+                    }
+                    else if (Character.isDigit(ch))
+                    {
+                      // Accumulate the value as it's being read in order
+                      maskValue = maskValue * 10 + Character.getNumericValue(ch);
+                    }
+                    else
+                    {
+                      throw new IllegalMaskException("Malformed mask syntax: unexpected range value");
+                    }
+                    subPosition++;
+                  }
                 }
                 else
                 {
-                  throw new IllegalMaskException("Malformed mask syntax: unexpected range value");
-                }
-                ii++;
-              }
+                  if (toparse.charAt(subPosition + 1) != '(')
+                  {
+                    throw new IllegalMaskException("Malformed mask syntax: expected '(' token");
+                  }
 
-              // Set the mask value to the range value specified for the parameter
-              maskValue = Integer.valueOf(rangeValue.toString());
+                  state = ParseState.DESCEND;
+                  nextToken = subPosition;
+                }
+                break;
+              case ')':
+                state = ParseState.ASCEND;
+                nextToken = subPosition;
+                break;
+              default:
+                field.append(c);
```

Comment:
check if c is whitespace before appending, see my comment below.
field names cannot have spaces in-between, so we can ignore whitespaces.